orderbook/src/orderbook.rs

```diff
@@ -82,9 +82,21 @@ impl Orderbook {
             Some(order) => order,
             None => return Ok(AddOrderResult::InvalidSignature),
         };
-        self.balance_fetcher
-            .register(order.order_meta_data.owner, order.order_creation.sell_token)
-            .await;
+        let owner = order.order_meta_data.owner;
+        let token = order.order_creation.sell_token;
+        self.balance_fetcher.register(owner, token).await;
+        let minimum_balance = if order.order_creation.partially_fillable {
+            1.into()
+        } else {
+            order
+                .order_creation
+                .sell_amount
+                .saturating_add(order.order_creation.fee_amount)
+        };
+        let balance = self.balance_fetcher.get_balance(owner, token);
+        if !matches!(balance, Some(balance) if balance >= minimum_balance) {
```

Comment:
With this match statement, it's not clear to me what happens when balance is None.

Reply:
I read this as "if `balance` doesn't match `Some` balance that is bigger than the minimum". So a `None` balance would enter the if block (and return `InsufficientFunds` result).